tests/test-components/components/integration-spin-inbound-http/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+wit_bindgen::generate!({
+    world: "http-trigger",
+    path: "../../../../wit-0.2.0/deps/spin@unversioned",
+    exports: {
+        "fermyon:spin/inbound-http": SpinHttp,
+    }
+});
+
+use std::collections::HashMap;
+
+use exports::fermyon::spin::inbound_http::{self, Request, Response};
+use miniserde::json;
+
+struct SpinHttp;
+
+impl inbound_http::Guest for SpinHttp {
+    fn handle_request(req: Request) -> Response {
+        let (status, body) = match handle_request(req) {
+            Ok(body) => (200, body),
+            Err(err) => (500, format!("{err:?}").into_bytes()),
+        };
+        Response {
+            status,
+            headers: None,
+            body: Some(body),
+        }
+    }
+}
+
+fn handle_request(req: Request) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
+    if !req.params.is_empty() {
+        return Err("request params field is deprecated".into());
+    }
+
+    let headers = req.headers.into_iter().collect::<HashMap<_, _>>();
+    let path = headers
+        .get("spin-path-info")
+        .map(|path| path.as_str())
+        .unwrap_or("");
+    match path {
+        "/echo" => Ok(req.body.unwrap_or_default()),
```

Comment:
One thing I'd like for us to try to do is to remove redundancy in our testing. We have other tests that test echoing capabilities as well as tests that read request headers. The unique elements of this test seem to be asserting that `params` is empty and reading `spin-path-info`. I wonder if this test can be simplified?

Reply:
Don't the other tests only cover wasi handlers?

Reply:
Ah you're right. Ok - we should probably make tests that test the wasi handler and the Spin handler mirror one another so we can more easily see if there are gaps in test coverage, but that can be done in a follow up.